Fixed incorrect parsing of the failed LLVM/Clang Unit test names

slydiman opened the pull request:
https://github.com/llvm/llvm-zorg/issues/49

slydiman commented:
Note this patch updates the test name to something like `ADT/./ADTTests.exe/PagedVectorTest/ShrinkTest` but os.path.basename() keeps only the last part of the name after the last /.

![sample](https://github.com/llvm/llvm-zorg/assets/16115622/b5f5195c-bf0a-4cd9-afe3-13833bf24f5d)

vvereschaka reviewed (commented):
To make a log name more relative to the failed test I propose the following:
* move you lines 71-75 close to `addCompleteLog` call
* update the `addCompleteLog` call with the next changes
```python
          # Add the log to the build status.
          # Make the test name short, the qualified test name is in the log anyway.
          # Otherwise, we run out of the allowed name length on some hosts.
          name_part = name.rpartition('::')
          name_part2 = basename(name_part[2])
          
          if self.activeVerboseGTest:
              m = self.kTestVerboseGTestUnitRE.match(name.strip())
              if m:
                  name_part2 = '/'.join([name_part2,] + self.activeVerboseGTest)

          self.step.addCompleteLog(
                      code + ': ' + name_part[0].strip() + name_part[1] + name_part2),
                      '\n'.join(self.activeVerboseLog))
```
We should get something like that for the log file name `LLVM-Unit :: ADTTests.exe/PagedVectorTest/EmptyTest`.

slydiman commented on zorg/buildbot/commands/LitTestCommand.py:
```diff
@@ -16,6 +16,8 @@ class LitLogObserver(LogLineObserver):
   kTestVerboseLogStartRE = re.compile(r"""\*{4,80} TEST '(.*)' .*""")
   kTestVerboseLogStopRE = re.compile(r"""\*{10,80}""")
+  kTestVerboseGTestUnitRE = re.compile(r'^(.*)\/(\d*)\/(\d*)$')
+  kTestVerboseGTestPathRE = re.compile(r'^(.+) --gtest_filter=(\w+).(\w+)$')
```
`^(.+)\s+--gtest_filter=(\w+).(\w+)[\s$]` will not work because `[\s$]` means a space or $ sign, not the EOL.

I will update the regex with the second suggestion.

vvereschaka replied:
`\s$` - space or EOL
`\s\$` - space or `$` sign
but anyway, I'm ok with any of those regexes.

it should work. I have tested it here https://regex101.com/

slydiman replied:
https://regex101.com/r/m187iA/1

vvereschaka replied:
You didn't add EOL to the test string there; otherwise it gets matched.

slydiman replied:
Note `^(.+)\s+--gtest_filter=(\w+).(\w+)(\s|$)` or even `^(.+)\s+--gtest_filter=(\w+).(\w+)$` work without `\n` at the end.

Anyway outLineReceived(self, line) receives the line without `\n` at the end. Look at _lineReceived() in buildbot/master/buildbot/process/logobserver.py

slydiman commented:
>```
># name = `LLVM-Unit :: ADT/./ADTTests.exe/4/93`
>name_part = name.rpartition('::') # ['LLVM-Unit ', '::', ' ADT/./ADTTests.exe/4/93']
>name_part2 = basename(name_part[2]) # name_part2 = '93'
>          
>if self.activeVerboseGTest:
>  m = self.kTestVerboseGTestUnitRE.match(name.strip())
>  if m:
>      name_part2 = '/'.join([name_part2,] + self.activeVerboseGTest)
>      # name_part2 = '93/PagedVectorTest/EmptyTest'
>
>self.step.addCompleteLog(
>    code + ': ' + name_part[0].strip() + name_part[1] + name_part2),
>    # FAILED: LLVM-Unit::93/PagedVectorTest/EmptyTest
>    '\n'.join(self.activeVerboseLog))
>```
>
> We should get something like that for the log file name `LLVM-Unit::ADTTests.exe/PagedVectorTest/EmptyTest`.

I have updated this code with the following
```
# name = `LLVM-Unit :: ADT/./ADTTests.exe/4/93`
name_part = name.rpartition('::') # ['LLVM-Unit ', '::', ' ADT/./ADTTests.exe/4/93']
name_part2 = basename(name_part[2]) # name_part2 = '93'

if self.activeVerboseGTest:
    m = self.kTestVerboseGTestUnitRE.match(name_part[2].strip()) # 'ADT/./ADTTests.exe/4/93'
    if m:
        name_part2 = '/'.join([basename(m.group(1)),] + self.activeVerboseGTest)
        # name_part2 = 'ADTTests.exe/PagedVectorTest/EmptyTest'

self.step.addCompleteLog(
    code + ': ' + name_part[0].strip() + name_part[1] + name_part2),
    # FAILED: LLVM-Unit::ADTTests.exe/PagedVectorTest/EmptyTest
    '\n'.join(self.activeVerboseLog))
```
Note I have also updated the rexeg with `^(.*)\/(\d+)\/(\d+)$`.

vvereschaka reviewed (approved):
LGTM

Did you get a screenshot how it looks with these changes?

slydiman commented:
> Did you get a screenshot how it looks with these changes?

![sample](https://github.com/llvm/llvm-zorg/assets/16115622/d48c1721-a178-404d-8db8-a2ee372af022)

gkistanova reviewed (approved):
LGTM.
